syntax = "proto3";

package bondPackage;

service BondServer {
  rpc load(LoadRequest) returns (LoadResponse) {}
  rpc predict(InputRequest) returns (InputResponse) {}
}

message LoadRequest {
  string bitfileName = 1;
  string hlsToolkit  = 2;
  string targetBoard = 3;
  string projectType = 4;
  int32  nInputs = 5;
  int32  nOutputs = 6;
  string flavor = 7;
  string sourceNeuralNetwork = 8;
}

message LoadResponse {
  bool success = 1;
  string message = 2;
}

message InputRequest {
  string inputs = 1;
}

message InputResponse {
  bool success = 1;
  string outputs = 2;
}